File: app/src/main/java/temple/edu/yelp_randomizer/models/ReviewsModel.java
package temple.edu.yelp_randomizer.models;

import android.os.Parcel;
import android.os.Parcelable;

public class ReviewsModel implements Parcelable {

    int rating;
    String text;
    String ProfileImageURL;
    String userName;

    public ReviewsModel(int rating, String text, String profileImageURL, String userName) {
        this.rating = rating;
        this.text = text;
        ProfileImageURL = profileImageURL;
        this.userName = userName;
    }

    protected ReviewsModel(Parcel in) {
        rating = in.readInt();
        text = in.readString();
        ProfileImageURL = in.readString();
        userName = in.readString();
    }

    public static final Creator<ReviewsModel> CREATOR = new Creator<ReviewsModel>() {
        @Override
        public ReviewsModel createFromParcel(Parcel in) {
            return new ReviewsModel(in);
        }

        @Override
        public ReviewsModel[] newArray(int size) {
            return new ReviewsModel[size];
        }
    };

    public int getRating() {
        return rating;
    }

    public void setRating(int rating) {
        this.rating = rating;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getProfileImageURL() {
        return ProfileImageURL;
    }

    public void setProfileImageURL(String profileImageURL) {
        ProfileImageURL = profileImageURL;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(rating);
        parcel.writeString(text);
        parcel.writeString(ProfileImageURL);
        parcel.writeString(userName);
    }
}

File: app/src/main/java/temple/edu/random/models/ReviewsModel.kt
package temple.edu.random.models

import android.os.Parcelable
import android.os.Parcel
import android.os.Parcelable.Creator

class ReviewsModel : Parcelable {
    var rating: Int
    var text: String?
    var profileImageURL: String?
    var userName: String?

    constructor(rating: Int, text: String?, profileImageURL: String?, userName: String?) {
        this.rating = rating
        this.text = text
        this.profileImageURL = profileImageURL
        this.userName = userName
    }

    protected constructor(`in`: Parcel) {
        rating = `in`.readInt()
        text = `in`.readString()
        profileImageURL = `in`.readString()
        userName = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(rating)
        parcel.writeString(text)
        parcel.writeString(profileImageURL)
        parcel.writeString(userName)
    }

    companion object {
        val CREATOR: Creator<ReviewsModel?> = object : Creator<ReviewsModel?> {
            override fun createFromParcel(`in`: Parcel): ReviewsModel? {
                return ReviewsModel(`in`)
            }

            override fun newArray(size: Int): Array<ReviewsModel?> {
                return arrayOfNulls(size)
            }
        }
    }
}